compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1765,7 +1765,7 @@ class Typer(@constructorOnly nestingLevel: Int = 0) extends Namer
           else report.error(new DuplicateBind(b, cdef), b.srcPos)
           if (!ctx.isAfterTyper) {
             val bounds = ctx.gadt.fullBounds(sym)
-            if (bounds != null) sym.info = bounds
+            if (bounds != null) sym.info = checkNonCyclic(sym, bounds, reportErrors = true)
```

Comment:
Is checkNonCyclic still needed with the other changes?

Reply:
No, but I thought it would be good to have, to avoid ending up with cyclic bounds.  I can leave a comment saying so, or do you think we shouldn't run it?

Reply:
If we can't think of a situation where the check would trigger I would rather leave it out I think.